packages/optimizer/lib/transformers/AmpScriptCsp.js

```diff
@@ -0,0 +1,90 @@
+/**
+ * Copyright 2019 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+'use strict';
+
+const {calculateHash} = require('@ampproject/toolbox-script-csp');
+
+/**
+ * AmpScriptCsp - adds CSP for amp-script.
+ *
+ * Currently only supports inline scripts.
+ *
+ * This transformer supports the following parameter:
+ *
+ * `append`: specifies the calculated CSPs should be appended to existing ones.
+ */
+class AmpScriptCsp {
+  transform(tree, params) {
+    const html = tree.root.firstChildByTag('html');
+    const head = html.firstChildByTag('head');
+    const body = html.firstChildByTag('body');
+
+    if (!head || !body) return;
+
+    const inlineScripts = this._findAllInlineScripts(body);
+    const hashes = new Set();
+    for (const script of inlineScripts) {
+      const content = script.children[0].data;
+      hashes.add(calculateHash(content));
+    }
+
+    const cspMeta = this._findOrCreateCspMeta(tree, head);
```

Comment:
nit: I'd first add the existing hashes. This makes it clearer that optimizer added new hashes. Also, you can use `new Set(existingCsp)` instead of the loop.

Reply:
`new Set(existingCsp)` wouldn't work, I'd just have two sets then (and there seems to be no way to append an array to a Set). But after I moved existing hashes to the top, I was able to get rid of the loop!